lib/rules/no-extra-parens.js

```diff
@@ -710,6 +710,20 @@ module.exports = {
             reportsBuffer.reports = reportsBuffer.reports.filter(r => r.node !== node);
         }
 
+        /**
+         * Checks whether a node is a MemberExpression at NewExpression's callee.
+         * @param {ASTNode} node node to check.
+         * @returns {boolean} True if the node is a MemberExpression at NewExpression's callee. false otherwise.
+         */
+        function isMemberExpInNewCallee(node) {
+            if (node.type === "MemberExpression") {
+                return node.parent.type === "NewExpression" && node.parent.callee === node
+                    ? true
+                    : isMemberExpInNewCallee(node.parent);
+            }
+            return false;
+        }
```

Comment:
I think this should go up the tree only if it's `object` child of another `MemberExpression`, otherwise this would be a false negative:

```js
"new a[(b()).c]"
```

Reply:
@mdjermanovic 
Thanks! 👍  I added a checking about that and test case.